core/cowdata.h

```diff
@@ -280,6 +323,13 @@ Error CowData<T>::resize(int p_size) {
 			_ptr = (T *)ptr;
 
 		} else {
+			if (r_elem_backup) {
+				const T *original = r_elem_backup->get_original_ptr();
+				if (original >= _ptr && original < _ptr + size()) {
+					r_elem_backup->make_backup();
+				}
+			}
+
```

Comment:
Just to note, this runs fine on mine, but _currently_ with vector push_back, I believe it is calling realloc on every push_back (except the first element), and relying on realloc to detect a similar allocation size (power of 2). This in turn means that this if statement will always evaluate to true and it will always copy by value.